src/Datadog.Trace/Span.cs

```diff
@@ -138,23 +139,47 @@ public Span SetTag(string key, string value)
 
             if (value == null)
             {
+                // Agent doesn't accept null tag values,
+                // remove them instead
                 Tags.TryRemove(key, out _);
+                return this;
             }
-            else
+
+            // some tags have special meaning
+            switch (key)
```

Comment:
Similar feedback here as to other PRs - we seem to just really want to keep piling the logic/responsibilities inside a single class (Span in this case) when the responsibility here is external, and could easily be separated away.  I would particularly at this point start pulling this stuff out into an external tagging service(s) that handle this....also starts/moves us down a path towards making a span a dumb dto that is managed by a external composable things....